File: 344. Reverse String AC.cpp
#include <string>

using namespace std;

class Solution
{
public:
    string reverseString(string s)
    {
        string reverse;

        // NOTE(Wuxiang): Reverse iteration with size_t
        for (size_t i = s.length() - 1; i < s.length(); --i)
        {
            reverse.push_back(s[i]);
        }

        return reverse;
    }
};

File: 344. Reverse String AC.cs

    public class Solution
    {
        public string ReverseString(string s)
        {
            var reverse = s.Reverse();
            var reverseBuilder = new StringBuilder();
            foreach (var c in reverse)
            {
                reverseBuilder.Append(c);
            }

            return reverseBuilder.ToString();
        }
    }
